plugins/woocommerce/src/StoreApi/Authentication.php

```diff
@@ -232,6 +232,10 @@ protected static function get_ip_address( bool $proxy_support = false ) {
 			return self::validate_ip( sanitize_text_field( wp_unslash( $_SERVER['REMOTE_ADDR'] ?? 'unresolved_ip' ) ) );
 		}
 
+		if ( array_key_exists( 'HTTP_CF_CONNECTING_IP', $_SERVER ) ) {
```

Comment:
Note that using this header is fine for geolocation, but probably can't be used here, without also checking whether the request is also coming form one of [CloudFlare's IP address](https://www.cloudflare.com/en-gb/ips/)